processor-api.md

```diff
@@ -0,0 +1,85 @@
+# Processor API
+
+A Processor is a class with static methods that is responsible for type specific logic to convert from an instance into a template and then back to an instance.
+
+## Determining if the processor should be used
+
+Separate creation from deployment for typing
+
+
+
+```js
+public static canConvertToTemplate(
+    entity: IItem || IGroup
+  ): Promise<boolean> {...}
+
+public static canDeployTemplate(
+    template: ITemplate
+  ): Promise<boolean> {...}
+```
+
+Given an IItem, the processor can inspect the type, typekeywords or anything else about the item to determine if it should process it. Currently this is done via a simple lookup based on `item.type`. This new pattern inverts the problem while also allowing the processor itself to inspect the item and determine if it should process it.
+
+
+## License & Privilege Checking
+
+Used during the deployment process, before starting the actual deployment of template, we run a check if the current user has the licensing / privileges required to deploy the template.
+
+```js
+public static canUserDeployTemplate(
+  currentUser: IUser, // IUser is imported from REST-JS,
```

Comment:
why not just `user` - as could inquire about another User which may not be the `current` user